circus-train-hive-view/src/main/java/com/hotels/bdp/circustrain/hive/view/transformation/TableProcessor.java

```diff
@@ -35,12 +35,15 @@
   @Override
   public Object process(Node node, Stack<Node> stack, NodeProcessorCtx procCtx, Object... nodeOutputs)
     throws SemanticException {
-    ASTNode astNode = (ASTNode) node;
+    ASTNode astNode;
+    if (node instanceof ASTNode) {
+      astNode = (ASTNode) node;
+    } else {
+      throw new ClassCastException("Cannot cast Node to ASTNode");
+    }
     if (astNode.getToken() != null && astNode.getToken().getText() != null) {
-      switch (astNode.getToken().getText()) {
-      case "TOK_TABNAME":
+      if ("TOK_TABNAME".equals(astNode.getToken().getText())) {
         tables.add(extractTableName(astNode));
-        break;
```

Comment:
This was a switch statement with one case and no `default`, so it was replaced by an if statement.

Reply:
OK, maybe the intention was to support more later, what you've done is fine, it can always be converted back later if more token types are needed.